huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_custom_event_channel_test.go

```diff
@@ -0,0 +1,201 @@
+	client, err := cfg.EgV1Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating EG v1 client: %s", err)
+	}
+
+	return custom.Get(client, state.Primary.ID, state.Primary.Attributes["enterprise_project_id"])
+}
+
+// Without enterprise project association (Notes: not default enterprise project)
+func TestAccCustomEventChannel_basic(t *testing.T) {
+	var (
+		obj custom.Channel
+
+		rName = "huaweicloud_eg_custom_event_channel.test"
+		name  = acceptance.RandomAccResourceName()
+		rc    = acceptance.InitResourceCheck(rName, &obj, getCustomEventChannelFunc)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCustomEventChannel_basic_step1(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttr(rName, "description", "Created by acceptance test"),
+					resource.TestCheckResourceAttrSet(rName, "created_at"),
+				),
+			},
+			{
+				Config: testAccCustomEventChannel_basic_step2(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttr(rName, "description", ""),
+					resource.TestCheckResourceAttr(rName, "cross_account_ids.#", "2"),
+					resource.TestCheckResourceAttr(rName, "cross_account_ids.0", "account1"),
+					resource.TestCheckResourceAttr(rName, "cross_account_ids.1", "account2"),
+					resource.TestCheckResourceAttrSet(rName, "updated_at"),
+				),
+			},
+			{
+				// Restored test.
+				Config: testAccCustomEventChannel_basic_step3(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttr(rName, "description", "Created by acceptance test"),
+					resource.TestCheckResourceAttr(rName, "cross_account_ids.#", "0"),
+					resource.TestCheckResourceAttrSet(rName, "updated_at"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCustomEventChannel_basic_step1(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_eg_custom_event_channel" "test" {
+  name        = "%[1]s"
+  description = "Created by acceptance test"
+}
+`, name)
+}
+
+func testAccCustomEventChannel_basic_step2(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_eg_custom_event_channel" "test" {
+  name              = "%[1]s"
+  cross_account_ids = ["account1", "account2"]
+}
+`, name)
+}
+
+func testAccCustomEventChannel_basic_step3(name string) string {
```

Comment:
I think you can just use function `testAccCustomEventChannel_basic_step1`, and remove `testAccCustomEventChannel_basic_step3`